analyses/printPlots_run.C

```diff
@@ -0,0 +1,1050 @@
+//#include "tdrstyle.C"
+#include "TH1.h"
+#include "TH2.h"
+#include <TStyle.h>
+#include <TCanvas.h>
+#include "TGraph.h"
+#include "TCanvas.h"
+#include "TAxis.h"
+#include "TStyle.h"
+#include "TLegend.h"
+#include "TGraphAsymmErrors.h"
+#include "PlotTemplate.C"
+
+double getMeanEfficiency(TH1 *plot1, TH1 *plot2, double initialVal, double finalVal){
+  double sumPassed = 0.;
+  double sumTotal  = 0.;
+  for (unsigned int i = 1; i<=plot1->GetXaxis()->GetNbins(); i++){
+     if ( plot1->GetXaxis()->GetBinCenter(i) < initialVal || plot1->GetXaxis()->GetBinCenter(i) > finalVal  ) continue; 
+     sumPassed = sumPassed + plot1->GetBinContent(i); 
+     sumTotal  = sumTotal  + plot2->GetBinContent(i); 
+  //   cout << plot1->GetXaxis()->GetBinCenter(i) << " " <<sumPassed << "/" << sumTotal << endl;  
+  }
+  
+  return sumPassed/sumTotal;
+}
+
+void remove_x_errors(TGraphAsymmErrors *plot) {
+  for (int i = 0; i < plot->GetXaxis()->GetNbins(); i++) {
+    plot->SetPointEXlow(i, 0);
+    plot->SetPointEXhigh(i, 0);
+  }
+  plot->SetTitle("");
+  // plot->SetMarkerStyle(20);
+  // plot->SetMarkerSize(20);
+}
+
+void printPlots_run(std::string run) {
+  std::cout << "Beginning..." << std::endl;
+  const bool fileOK = false; 
+
+  // gStyle->SetOptStat(111111);
+  gStyle->SetOptStat(0);
+  std::vector<std::string> stuffTags = {"Time", "Pos", "Psi"};
+
+  TString runNumber = run; 
+  //TString file = "results_run" + runNumber + ".root";
+  TString file = "run" + runNumber + "/emulator/results_emulator.root";
+  //gSystem->Exec("mkdir run" + runNumber);
+
+  TFile data1(file);
+  
+  TString Lumi = "35.9";
```

Comment:
Isn't it annoying to keep this TString in the middle of the code, if we want to change it in the future?

Reply:
fixed